src/routes/CreateSafePage/steps/OwnersAndConfirmationsNewSafeStep.tsx

```diff
@@ -132,12 +133,20 @@ function OwnersAndConfirmationsNewSafeStep(): ReactElement {
       <Hairline />
       <Block margin="md" padding="md">
         <RowHeader>
-          {owners.map(({ nameFieldName, addressFieldName }) => {
+          {owners.map(({ nameFieldName, addressFieldName }, i: number) => {
             const hasOwnerAddressError = formErrors[addressFieldName]
             const ownerAddress = createSafeFormValues[addressFieldName]
             const showDeleteIcon = addressFieldName !== 'owner-address-0' // we hide de delete icon for the first owner
             const ownerName = ownersWithENSName[ownerAddress] || 'Owner Name'
 
+            const isRepeated = (value: string) => {
+              const prevOwners = owners.slice(0, i)
```

Comment:
We should check in both directions for the validation in case a previous address field is changed. Using filter would cover this i.e. `const prevOwners = owners.filter((owner, index) => index !== i)`